bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,239 @@
+            return false;
+        }
+        if (response == null) {
+            logger.debug("Command Status returned null");
+            device = null;
+            updateStatus(ThingStatus.OFFLINE);
+            return false;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        return true;
+    }
+
+    private void startUpdateJob() {
+        logger.debug("Starting Etherrain Update Job");
+        this.updateJob = scheduler.scheduleWithFixedDelay(this::updateBridge, 0, config.refresh, TimeUnit.SECONDS);
+
+        logger.debug("EtherRain sucessfully initialized. Starting status poll at: {}", config.refresh);
+    }
+
+    private void stopUpdateJob() {
+        logger.debug("Stopping Etherrain Update Job");
+
+        final ScheduledFuture<?> updateJob = this.updateJob;
+        if (updateJob != null && !updateJob.isDone()) {
+            updateJob.cancel(false);
+        }
+
+        this.updateJob = null;
+    }
+
+    private boolean updateBridge() {
+        if (!connected || device == null) {
+            connected = connectBridge();
+            if (!connected || device == null) {
+                connected = false;
+                device = null;
+                logger.debug("Could not connect to Etherrain device.");
+                return false;
+            }
+        }
+
+        EtherRainStatusResponse response;
+
+        try {
+            response = device.commandStatus();
+        } catch (EtherRainException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the EtherRain");
+            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}", e.toString());
+            device = null;
+            return false;
+        }
+
+        switch (response.getOperatingStatus()) {
+            case STATUS_READY:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("READY"));
+                break;
+            case STATUS_WAITING:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("WAITING"));
+                break;
+            case STATUS_BUSY:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("BUSY"));
+                break;
+        }
+
+        switch (response.getLastCommandStatus()) {
+            case STATUS_OK:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("OK"));
+                break;
+            case STATUS_ERROR:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("ERROR"));
+                break;
+            case STATUS_UNATHORIZED:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("UNATHORIZED"));
+                break;
+        }
```

Comment:
If you removed the STATUS_ prefix from these enum names you could simplify all this with:
```suggestion
        updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType(response.getLastCommandStatus().name()));
```